### ISSUE TITLE
Items display twice when scanned on the soda machine.

### ISSUE THREAD

**Issue by thalvors** [chezbob/chezbob-legacy#34]
Lonnie (user lonnie) bought two caffeine free diet cokes by scanning them at the soda machine.  Two of them show up in his history, but the soda machine displayed as if he bought 4.  I am guessing it only deducted the cost of 2 from his balance but have no evidence of this.  We should make sure it's not deducting double balance, and then make it not show up twice.

**Comment by d1m0** [comment]
This is a UI problem only. Users don't get charged double. I suspect its caused by the code in soda-common.js that reconfigures EventSource upon disconnect:

```
var source = new EventSource('/stream');
source.onerror = function(e){
    //display nice error message
    $("#disconnected").modal('show').on('shown.bs.modal', function(e) {
        setTimeout(function() {
            $("#disconnected").modal('hide').on('hidden.bs.modal', function(e) {configureEventSource();});
        }
        , 15000);});
}
```

More specifically I suspect we either double-subscribe to /stream or the event handler fires twice somehow.

**Comment by d1m0** [comment]
The following should fix this (assuming we have the correct root cause):

commit 7de5f4ae75d4723b6b89c17bf959be842dc73dcb
Author: Dimitar Bounov dimo@ucsd.edu
Date:   Wed Sep 10 16:44:59 2014 -0700

```
Potential fix for items appearing twice in the soda menu
```

diff --git a/bob2k14/static/js/soda-common.js b/bob2k14/static/js/soda-common.js
index 3cc6046..bcfb5e4 100644
--- a/bob2k14/static/js/soda-common.js
+++ b/bob2k14/static/js/soda-common.js
@@ -143,10 +143,12 @@ function refreshsodastates()
        }
 }

+window.source = null;
+
 function configureEventSource()
 {
-    var source = new EventSource('/stream');
-    source.onerror = function(e){
-    window.source = new EventSource('/stream');
-    window.source.onerror = function(e){
       //display nice error message
       $("#disconnected").modal('show').on('shown.bs.modal', function(e) {
           setTimeout(function() {
  @@ -154,7 +156,7 @@ function configureEventSource()
           }
           , 15000);});
   }
-       source.onmessage = function(e) {
-       window.source.onmessage = function(e) {
              if (e.data.substring(0,3) == "sbc")
              {
                      //this is a barcode that was purchased

Basically lets make the source variable global, so no matter how many times configureEventSource is called, there is always a single source.
